inspec/a2-iam-v2-upgrade-legacy-v1-policies/controls/base.rb

```diff
@@ -0,0 +1,322 @@
+        "/api/v0/auth/users/#{NON_ADMIN_USERNAME_V2}",
+        http_method: 'DELETE',
+      )
+      expect(delete_non_admin_request.http_status.to_s).to match(/200|404/)
+
+      delete_token_request = automate_api_request(
+        "/api/v0/auth/tokens/#{TEST_TOKEN_ID_V2}",
+        http_method: 'DELETE',
+      )
+      expect(delete_token_request.http_status.to_s).to match(/200|404/)
+    end
+
+    # compliance:profiles for client calls should be permitted by default
+    describe '/api/v0/compliance/profiles/search as client' do
+      it "api/v0/compliance/profiles/search returns the correct response code for client" do
+        expect(
+          automate_client_api_request(
+            "/api/v0/compliance/profiles/search",
+            TEST_TOKEN_V2,
+            http_method: 'POST'
+          ).http_status
+        ).to eq(200)
+      end
+    end
+
+    describe 'legacy data-collector endpoints' do
+      %w(
+        /api/v0/events/data-collector
+        /data-collector/v0
+      ).each do |url|
+        { 'GET': 200, 'POST': 400 }.each do |method, status|
+          it "#{method} #{url} returns the correct response code for client" do
+            expect(
+              automate_client_api_request(
+                url,
+                TEST_TOKEN_V2,
+                http_method: method,
+              ).http_status
+            ).to eq(status)
+          end
+        end
+      end
+    end
+
+    describe '/api/v0/ingest/events/chef/' do
+      {
+        run: { entity_uuid: 'a6597b02-2d83-47ce-8e46-84d1e85be6c7' },
+        action: {},
+        nodedelete: { node_id: 'a6597b02-2d83-47ce-8e46-84d1e85be6c7' },
+      }.each do |url, body|
+        it "#{url} returns the correct response code for client" do
+          expect(
+            automate_client_api_request(
+              "/api/v0/ingest/events/chef/#{url}",
+              TEST_TOKEN_V2,
+              http_method: 'POST',
+              request_body: body.to_json,
+            ).http_status
+          ).not_to eq(403)
+        end
+      end
+    end
+
+    # ingest multiple delete
+    describe '/api/v0/ingest/events/chef/node-multiple-deletes' do
+      it "returns the correct response code for client" do
+        expect(
+            automate_client_api_request(
+              "/api/v0/ingest/events/chef/node-multiple-deletes",
+              TEST_TOKEN_V2,
+              http_method: 'POST',
+              request_body: { node_ids: [ 'fake-2d83-47ce-8e46-84d1e85be6c7' ] }.to_json,
+            ).http_status
+          ).to eq(403)
+      end
+    end
+
+    shared_examples 'authz access control (iam v2)' do
+      # ingest
+      describe '/api/v0/ingest/events/chef/' do
+        {
+          'run': { 'entity_uuid': 'a6597b02-2d83-47ce-8e46-84d1e85be6c7' },
+          'action': {},
+          'nodedelete': { 'node_id': 'a6597b02-2d83-47ce-8e46-84d1e85be6c7' },
+        }.each do |url, body|
+          it "#{url} returns the correct response code for user" do
```

Comment:
should we specify `not 403` instead of `correct response code` in these test names?